### ISSUE TITLE
Infinite animateTo loop when reaching onEnd

### ISSUE THREAD

**Issue by rienkkk** [MSVCode/scroll_snap_list#23]
With the following setup I get an infinity loop when reaching the end of the list.

```
 ScrollSnapList(
      selectedItemAnchor: SelectedItemAnchor.START,
      itemSize: 440,
      itemBuilder: (BuildContext context, int index) {
        return XCard(routine: x[index]);
      },
      itemCount: x.length,
      onItemFocus: (int index) {
        HapticFeedback.lightImpact();
      },
      updateOnScroll: true,
      focusOnItemTap: false,
    ),
```

XCard is just a container with a width of 440.

The infinity loop results in the fact that the app crashes over time when staying at the end and that onTap gestures are not possible when reaching the end (because the listView thinks its animating, and won't accept onTap gestures). 

To fix this I added a onReachEnd check before _animateScroll. So if the end is reached it will not call animateScroll.

```
//only animate if not yet snapped (tolerance 0.01 pixel)
if ((scrollInfo.metrics.pixels - offset).abs() > 0.01) {
  _animateScroll(offset);
}
```

to

```
//only animate if not yet snapped (tolerance 0.01 pixel)
if ((scrollInfo.metrics.pixels - offset).abs() > 0.01) {
  if (!(scrollInfo.metrics.pixels >=
      scrollInfo.metrics.maxScrollExtent - tolerance)) {
    _animateScroll(offset);
  }
}
```

So I would suggest a simple on reach end check to prevent an infinite loop when reaching onEnd.

**Comment by Rokasurba** [comment]
I'm having the same issue. The list view keeps bouncing back and fourth when reaching the end.

**Comment by MSVCode** [comment]
Added in `0.8.5`

**Comment by rienkkk** [comment]
Good and bad news. The solution I provided was flawed. On some occasions, a list will not snap to the last item. Since then I have implemented a new solution, which is adding this lines to the NotificationListeners:

`      if (scrollInfo.depth > 0) {
          return false;
        }
`

It is quite a long time ago since I created this fix, but I could find no downside to this solution.

**Comment by MSVCode** [comment]
Thank you for the info & possible fix, changed in `0.8.6`
